aws_xray_sdk/core/models/dummy_entities.py

```diff
@@ -79,9 +85,12 @@ class DummySubsegment(Subsegment):
     to a dummy subsegment becomes no-op. Dummy subsegment will not
     be sent to the X-Ray daemon.
     """
-    def __init__(self, segment, name='dummy'):
 
+    def __init__(self, segment, name='dummy'):
         super(DummySubsegment, self).__init__(name, 'dummy', segment)
+        noOpID = os.getenv('AWS_XRAY_NOOP_ID')
+        if not noOpID or noOpID.lower() == 'true':
```

Comment:
Same here for checking for `false` instead of `true`. Also, don't we need an else statement? Otherwise the original subsegment ID is always generated.

Reply:
This [line](https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/subsegment.py#L98) calls the entity object and generates the random entity id (segment id). So basically that is why I have added env var checking condition in `DummySubSegment` class so that it initializes entity id (segment id) with `000000000`. So basically when env var value is `true` or not set which is the default case we generate entity id with `000000000` otherwise this line https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/subsegment.py#L98 will generate secure random entity id. Also, this condition https://github.com/aws/aws-xray-sdk-python/pull/293/files#diff-18ac8509f2bfc4403d165a3517044c0dec95f4c1f52a002fc4f6564cfcf351bdR31  will help us achieve that. So probably that is why we don't need checking for `false` condition.

Reply:
There will be 3 cases. 1, env var value `true` 2, env var value `false` 3, default. So, case 1 and 3 covered in this condition and for case 2 only we would generate secure random id.